mephisto/abstractions/providers/prolific/prolific_datastore.py

```diff
@@ -0,0 +1,390 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import sqlite3
+import threading
+from typing import Any
+from typing import Dict
+from typing import Optional
+
+from mephisto.abstractions.databases.local_database import is_unique_failure
+from mephisto.abstractions.providers.prolific.provider_type import PROVIDER_TYPE
+from mephisto.utils.logger_core import get_logger
+from . import api as prolific_api
+
+CREATE_REQUESTERS_TABLE = """
+CREATE TABLE IF NOT EXISTS requesters (
+    requester_id TEXT PRIMARY KEY UNIQUE,
+    is_registered BOOLEAN
+);
+"""
+
+CREATE_UNITS_TABLE = """
+CREATE TABLE IF NOT EXISTS units (
+    unit_id TEXT PRIMARY KEY UNIQUE,
+    is_expired BOOLEAN
+);
+"""
+
+CREATE_WORKERS_TABLE = """
+CREATE TABLE IF NOT EXISTS workers (
+    worker_id TEXT PRIMARY KEY UNIQUE,
+    is_blocked BOOLEAN
+);
+"""
```

Comment:
I don't think we'll need a workers table, no real worker-specific information to store (outside of qualifications, which fit in the main DB)

Reply:
Definitely, in that case we'll just use the local blocklist.